src/main/java/pl/javastart/task/TelephoneStore.java

```diff
@@ -0,0 +1,66 @@
+package pl.javastart.task;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Scanner;
+
+public class TelephoneStore extends Telephone {
+    Scanner sc = new Scanner(System.in);
+
+    private static final int MAX_TELEPHONES = 2;
+    private int telephoneIndex;
+    private Telephone[] telephones = new Telephone[MAX_TELEPHONES];
+
+    void addTelephones() {
+        System.out.println("Podaj informację o trzech telefonach");
+        System.out.println("Podaj nazwę:");
+        String name1 = sc.nextLine();
+        System.out.println("Podaj ilość pamięci:");
+        int memory1 = sc.nextInt();
+        sc.nextLine();
+        telephoneIndex++;
+        while (telephoneIndex < MAX_TELEPHONES) {
+            System.out.println("Podaj nazwę:");
+            String name2 = sc.nextLine();
+            System.out.println("Podaj ilość pamięci:");
+            int memory2 = sc.nextInt();
+            sc.nextLine();
+            if (name1.equals(name2)) {
+                if (memory1 == memory2) {
+                    System.out.println("Duplikat");
+                    telephoneIndex--;
+                }
+            }
+            telephoneIndex++;
+        }
+    }
```

Comment:
Tutaj brakuje tworzenia obiektów typu `Telephone`.
Algorytm powinien być taki, że w pętli:
- wczytujesz informacje o telefonie
- tworzysz obiekt Telephone
- sprawdzasz, czy takiego obiektu nie ma już w tablicy telephones
- jeżeli obiekt jest unikalny, to wstaw go na kolejne miejsce w tablicy  i zwiększ index o 1
- jeżeli obiekt już jest w tablicy, to niech pętla się powtarza bez zwiększenia indexu
Pętla powinna powtarzać się tak długo, dopóki index jest mniejszy od rozmiaru tablicy.

Najlepiej będzie wykorzystać pętlę do-while, bo będziemy wczytywali informacje o co najmniej jednym telefonie.
Sprawdzanie, czy obiekt jest w tablicy, czy nie sugeruję wydzielić do osobnej metody.